### PR TITLE
fix(range): enables range wheel event override

### PR DESCRIPTION
**Pull request by kid-icarus**
#### What does this PR do?

Enables a user to opt out of scroll wheel support inside the `RangeInput` component. This functionality was added in #5693 and has some buggy behavior as outlined in #7129. A native HTML range input does not allow the scroll wheel to change its underlying value. Here, we enable configuring this behavior in the theme.

**This is simply reopening a stale PR, and addressing PR feedback from it. Please see [the original PR](https://github.com/grommet/grommet/pull/6026) for more context**

#### Where should the reviewer start?

[This slack conversation](https://grommet.slack.com/archives/C09QQEG69/p1648134314121579) and then the RangeInput test file

#### What testing has been done on this PR?

Added tests for both the overridden and default scroll behavior

#### How should this be manually tested?

Manual testing can be done with the story in storybook

#### Do Jest tests follow these best practices?

- [x] `screen` is used for querying.
- [x] The correct query is used. (Refer to [this list of queries](https://testing-library.com/docs/queries/about/#priority))
- [x] `asFragment()` is used for snapshot testing.

#### Any background context you want to provide?

Just following up with https://github.com/grommet/grommet/pull/6026 since it has gone stale. Trying to get this across the finish line.

#### What are the relevant issues?

Closes https://github.com/grommet/grommet/issues/6025

#### Do the grommet docs need to be updated?

Yes

#### Should this PR be mentioned in the release notes?

Yes

#### Is this change backwards compatible or is it a breaking change?

Backwards compatible

### CODE REVIEW

**Review by ShimiSun — APPROVED** [review]
Thank you @kid-icarus for getting this PR to the finish line! 
The fix, tests, and TS adjustments look good, and thanks for adding descriptive code comments for context 🙌 

@jcfilben I believe you originally approved [the original PR](https://github.com/grommet/grommet/pull/6026#issuecomment-1940412182),
Ryan just incorporated the last feedback from @taysea to get it to the finish line.

File: src/js/components/RangeInput/RangeInput.js [hunk 1]
```diff
@@ -33,9 +36,12 @@ const RangeInput = forwardRef(
     },
     ref,
   ) => {
+    const theme = useContext(ThemeContext) || defaultProps.theme;
     const formContext = useContext(FormContext);
     const [focus, setFocus] = useState(focusProp);
 
+    const isScrollEnabled = theme?.rangeInput?.wheel !== false;
```
Comment (taysea) [inline src/js/components/RangeInput/RangeInput.js:43]:
```suggestion
    const scrollEnabled = theme?.rangeInput?.wheel !== false;
```

File: src/js/components/RangeInput/RangeInput.js [hunk 2]
```diff
@@ -49,13 +55,17 @@ const RangeInput = forwardRef(
 
     useEffect(() => {
       const { x, y } = scroll;
-      if (x !== null && y !== null) {
-        const handleScrollTo = () => window.scrollTo(x, y);
+      const handleScrollTo = () => window.scrollTo(x, y);
+      if (x !== null && y !== null && isScrollEnabled) {
```
Comment (taysea) [inline src/js/components/RangeInput/RangeInput.js:59]:
```suggestion
      if (x !== null && y !== null && scrollEnabled) {
```

File: src/js/components/RangeInput/RangeInput.js [hunk 3]
```diff
@@ -49,13 +55,17 @@ const RangeInput = forwardRef(
...
       }
-      return undefined;
-    }, [scroll]);
+      // there is no need to remove this event listener if scroll is disabled
+      // but we need to remove it if scroll is enabled and user switches to
+      // a theme with scroll disabled
+      return () => {
+        window.removeEventListener('scroll', handleScrollTo);
+      };
+    }, [scroll, isScrollEnabled]);
```
Comment (taysea) [inline src/js/components/RangeInput/RangeInput.js:68]:
```suggestion
    }, [scroll, scrollEnabled]);
```

File: src/js/components/RangeInput/RangeInput.js [hunk 4]
```diff
@@ -114,9 +124,9 @@ const RangeInput = forwardRef(
           setValue(event.target.value);
           if (onChange) onChange(event);
         }}
-        onMouseOver={handleMouseOver}
-        onMouseOut={handleMouseOut}
-        onWheel={handleOnWheel}
+        onMouseOver={isScrollEnabled ? handleMouseOver : undefined}
```
Comment (taysea) [inline src/js/components/RangeInput/RangeInput.js:127]:
```suggestion
        onMouseOver={scrollEnabled ? handleMouseOver : undefined}
```

File: src/js/components/RangeInput/RangeInput.js [hunk 5]
```diff
@@ -114,9 +124,9 @@ const RangeInput = forwardRef(
...
+        onMouseOut={isScrollEnabled ? handleMouseOut : undefined}
```
Comment (taysea) [inline src/js/components/RangeInput/RangeInput.js:128]:
```suggestion
        onMouseOut={scrollEnabled ? handleMouseOut : undefined}
```

File: src/js/components/RangeInput/RangeInput.js [hunk 6]
```diff
@@ -114,9 +124,9 @@ const RangeInput = forwardRef(
...
+        onWheel={isScrollEnabled ? handleOnWheel : undefined}
```
Comment (taysea) [inline src/js/components/RangeInput/RangeInput.js:129]:
```suggestion
        onWheel={scrollEnabled ? handleOnWheel : undefined}
```

**Review by taysea — COMMENTED** [review]
This looks good. One small suggestion re: variable naming to align with our conventions.

Additionally, can you update the PR description to better summarize the behavior? Thank you!

**Comment by kid-icarus** [comment]
> This looks good. One small suggestion re: variable naming to align with our conventions.
> 
> Additionally, can you update the PR description to better summarize the behavior? Thank you!

Done, thank you :)

**Review by taysea — COMMENTED** [review]
One final comment the I missed, can we add `wheel` in base.js as well. Typically when there's no value defined by default, we leave it commented out like this:

```
rangeInput: {
   // wheel: undefined,
}
```

**Comment by kid-icarus** [comment]
@taysea Done!

**Review by taysea — APPROVED** [review]
Thanks so much for being responsive to the feedback!

**Review by jcfilben — APPROVED** [review]
Looks good!

**Comment by jcfilben** [comment]
> We'd love to know when is the next planned release to incorporate this fix into our codebase.

Hey @ShimiSun I don't have exact timing on a release yet but most likely it will be either at the very end of this month or very beginning of March

**Pull request merged**